[CIAPP] add clarification on pipelines list (#14609)
DataDog/documentation: content/en/continuous_integration/troubleshooting.md

The default branch is used to power some features of the products, namely:

> Default branches list: This list only displays default branches. Setting the wrong default branch can result in missing or incorrect data in the default branches list.

Author: good catch, thanks @soyguijarro

> Default branches list: This list only displays default branches. Setting the wrong default branch can result in missing or incorrect data in the default branches list.

Reviewer: Also, should we say this default branches list is in the Tests page or something? It doesn't really say.

> - Tests Default branches list: This list only displays default branches. Setting the wrong default branch can result in missing or incorrect data in the default branches list.

Reviewer 2: ```suggestion
- Default branches list on the Tests page: This list only displays default branches. Setting the wrong default branch can result in missing or incorrect data in the default branches list.
```

> - Tests Wall Time comparison for non-default branches: in the Tests Branches list page, the **VS Default** column is calculated comparing wall time for the current branch against wall time for the default branch.

Reviewer 2: ```suggestion
- Wall time comparison for non-default branches: On the Tests page, in the Branches view, the **VS Default** column is calculated by comparing wall time for the current branch against wall time for the default branch.
```

- New flaky tests: Tests that are not currently classified as flaky in the default branch. If the default branch is not properly set, this could lead to a wrong number of detected new flaky tests.

> - Pipelines list: only default branches are displayed there, so a branch that is not results in a missing item from this table or the wrong branch being displayed.

Reviewer 2: ```suggestion
- Pipelines list: The pipelines list only displays default branches. Setting the wrong default branch can result in missing or incorrect data in the pipelines list.
```
I realize you copied this description from the "Default branches list" item above, but that one could also use improvement. If you'd like to improve the docs, you can update that item to the following:
- Default branches list: This list only displays default branches. Setting the wrong default branch can result in missing or incorrect data in the default branches list.

Author: Thanks for the review! Of course I want to improve the docs, applied your suggestion to both.
